### PR TITLE
Rwjs/unique sids

### PR DESCRIPTION
**Pull request by robsimmonds**
Create unique session ids even if reconnections are intermingled with new connections.

This builds on the session deletion PR from earlier today that I asked @confluence to review.

### CODE REVIEW

File: src/Session/SessionManager.cc [hunk 1]
```diff
@@ -51,9 +58,9 @@ void SessionManager::OnUpgrade(
         return;
     }
 
-    _session_number++;
-    // protect against overflow
-    _session_number = max(_session_number, 1u);
+    struct timeval tv;
```
Comment (veggiesaurus) [inline src/Session/SessionManager.cc]: we already use `uuid.h` for generating unique tokens. We shouldn't reinvent the wheel here

File: src/Session/SessionManager.cc [hunk 2]
```diff
@@ -97,6 +104,7 @@ void SessionManager::OnDisconnect(WSType* ws, int code, std::string_view message
     uint32_t session_id = static_cast<PerSocketData*>(ws->getUserData())->session_id;
 
     // Delete the Session
+    _sessions[session_id]->DecreaseRefCount();
```
Comment (veggiesaurus) [inline src/Session/SessionManager.cc]: I think we should first check `_sessions.count(session_id)`

**Review by veggiesaurus — CHANGES_REQUESTED** [review]

**Comment by veggiesaurus** [comment]
@robsimmonds does this mean we should close #1048 without merging?

**Comment by confluence** [comment]
> @robsimmonds does this mean we should close #1048 without merging?

@angus I think that we should first merge #1048, then merge dev into this, then merge this. I'm just waiting for CI to come back.

File: src/Session/SessionManager.cc (hunk 1 above)
Comment (robsimmonds) [reply src/Session/SessionManager.cc]: That would mean using 128 bit numbers for these ids. I guess we can do that, but it will need more testing.

Comment (veggiesaurus) [reply src/Session/SessionManager.cc]: the alternative would be to use `https://www.cplusplus.com/reference/chrono/high_resolution_clock/now/` I guess.

**Review by veggiesaurus — APPROVED** [review]

**Review by confluence — APPROVED** [review]

**Pull request merged**